src/Http/Routing.Abstractions/src/RoutingHttpContextExtensions.cs

```diff
@@ -46,8 +47,8 @@ public static object GetRouteValue(this HttpContext httpContext, string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
-            var routingFeature = httpContext.Features[typeof(IRoutingFeature)] as IRoutingFeature;
-            return routingFeature?.RouteData.Values[key];
+            return httpContext.Features.Get<IRoutingFeature>()?.RouteData.Values[key] ??
```

Comment:
@rynowak do we want this to work? Is this a legacy method?

Reply:
GetRouteValue/GetRouteData are legacy

Reply:
Does that mean GetRouteValue (which is visible by default) doesn't work when endpoint routing runs?

Reply:
It worked because the RouteData was created when requested and wrapped the route values - https://github.com/aspnet/AspNetCore/blob/master/src/Http/Routing/src/EndpointSelectorContext.cs#L43-L68

GetRouteValue should be changed to get the value from httpContext.Request.RouteValues to avoid allocating a RouteData.

Reply:
OK I made some changes.  The last thing to figure out is data tokens. I have some thoughts.